pkg/apis/core/validation/shoot_test.go

```diff
@@ -1806,6 +1806,77 @@ var _ = Describe("Shoot Validation Tests", func() {
 			}))))
 		})
 
+		It("should forbid enabling access control due to empty source field", func() {
+			shoot.Spec.Kubernetes.KubeAPIServer.AccessControl = &core.AccessControl{
+				Action: core.AuthorizationActionAllow,
+			}
+
+			errorList := ValidateShoot(shoot)
+
+			Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+				"Type":  Equal(field.ErrorTypeRequired),
+				"Field": Equal("spec.kubernetes.kubeAPIServer.accessControl.source"),
+			}))))
+		})
+
+		It("should forbid enabling access control due to source field containing no blocks", func() {
+			shoot.Spec.Kubernetes.KubeAPIServer.AccessControl = &core.AccessControl{
+				Action: core.AuthorizationActionAllow,
+				Source: core.AuthorizationSource{},
+			}
+
+			errorList := ValidateShoot(shoot)
+
+			Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+				"Type":   Equal(field.ErrorTypeRequired),
+				"Field":  Equal("spec.kubernetes.kubeAPIServer.accessControl.source"),
+				"Detail": Equal("source must not be empty, at least one block must be provided"),
+			}))))
+		})
+
+		It("should forbid enabling access control due to invalid IP Address", func() {
```

Comment:
I think it makes sense to add a test for the check for invalid CIDR.

Reply:
Added two more tests, one for invalid and one for valid CIDR.